### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
- [x] I have reviewed the [Contributor Guidelines][contributor].
- [x] I have reviewed the [Code of Conduct][coc] and agree to abide by it

[contributor]: https://github.com/bats-core/bats-core/blob/master/docs/CONTRIBUTING.md
[coc]:         https://github.com/bats-core/bats-core/blob/master/docs/CODE_OF_CONDUCT.md

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/bats-core/commit/cb356fd4056aae47ade81d52f1783caa4fc2681d#commitcomment-78360497

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/bats-core/commit/c843c0a7c629fa41eecdfd55db34ab64debb2c68

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           registry-url: "https://npm.pkg.github.com"
-      - name: scope package name as required by GHPR
+      - name: scope package name as required by GHCR
```
Comment (jsoref) [inline .github/workflows/release.yml:26]: ?

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ The format is based on [Keep a Changelog][kac] and this project adheres to
 * remaining instances of `run -<N>` being incorrectly documented as `run =<N>` (#599)
 * allow `--gather-test-outputs-in <directory>` to work with existing, empty
   directories (#603)
-  * also add `--clean-and-gather-test-outpust-in <directory>` for improved UX
+  * also add `--clean-and-gather-test-outputs-in <directory>` for improved UX
```
Comment (jsoref) [inline docs/CHANGELOG.md:31]: some projects don't like changing changelogs. Happy to drop

File: libexec/bats-core/bats-exec-suite [hunk 3]
```diff
@@ -7,7 +7,7 @@ num_jobs=${BATS_NUMBER_OF_PARALLEL_JOBS:-1}
 bats_no_parallelize_across_files=${BATS_NO_PARALLELIZE_ACROSS_FILES-}
 bats_no_parallelize_within_files=
 filter_status=''
-flags=('--dummy-flag') # add a dummy flag to prevent unset varialeb errors on empty array expansion in old bash versions
+flags=('--dummy-flag') # add a dummy flag to prevent unset variable errors on empty array expansion in old bash versions
```
Comment (jsoref) [inline libexec/bats-core/bats-exec-suite:10]: ?

File: test/load.bats [hunk 4]
```diff
@@ -8,7 +8,7 @@ setup() {
 }
 
 @test "find_in_bats_lib_path recognizes files relative to test file" {
-  test_dir="$BATS_TEST_TMPDIR/find_in_bats_lib_path/bats_test_dirname_priorty"
+  test_dir="$BATS_TEST_TMPDIR/find_in_bats_lib_path/bats_test_dirname_priority"
```
Comment (jsoref) [inline test/load.bats:11]: ?

**Review by jsoref — COMMENTED** [review]
Most corrections were automatically suggested by Google Sheets. All fault mine.

File: .github/workflows/release.yml (hunk 1 above)
Comment (martin-schulze-vireso) [reply .github/workflows/release.yml:26]: Maybe we should not use an acronym here. While ghcr seems to encompass npm, its core business is docker.

Comment (jsoref) [reply .github/workflows/release.yml:26]: Yes. GitHub mostly seems to talk about `GitHub Packages` in https://docs.github.com/en/packages

There's one user who they quote saying `GPR`,
and only two doc pages even yield search results for it, e.g. https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-net
and one for Ruby.
Google hits for GPR (in Canada?) generally are for `Ground Penetrating Radar` (and if you don't know why, you don't want to know).

**Review by martin-schulze-vireso — APPROVED** [review]

**Comment by martin-schulze-vireso** [comment]
Thanks for your contribution.

**Pull request merged**